app/src/ui/diff/index.tsx

```diff
@@ -952,6 +974,65 @@ export class Diff extends React.Component<IDiffProps, {}> {
     return null
   }
 
+  private renderLargeTextDiff() {
+    return (
+      <div className="panel empty large-diff">
+        <img src={NoDiffImage} />
+        <p>
+          The diff is too large to be displayed by default.
+          <br />
+          You can try to show it anyways, but performance may be negatively
+          impacted.
+        </p>
+        <Button onClick={this.onShowLargeDiff}>Show diff</Button>
+      </div>
+    )
+  }
+
+  private renderDiffTooLarge() {
+    return (
+      <div className="panel empty large-diff">
+        <img src={NoDiffImage} />
+        <p>The diff is too large to be displayed.</p>
+      </div>
+    )
+  }
+
+  private renderLargeText(diff: ILargeTextDiff) {
+    if (diff.hunks == null || diff.text == null) {
+      return fatalError('Cannot render diff withouth hunks and text')
```

Comment:
i'm not sure this fatalError will be needed if we follow my above suggestion but in case we keep it, typo: s/withouth/without